generators/base-application/application-config-all.d.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/consistent-type-imports */
+import type { Simplify } from 'type-fest';
+import type { ExportStoragePropertiesFromCommand } from '../../lib/command/types.js';
+import type { YO_RC_CONFIG_KEY } from '../../lib/utils/yo-rc.ts';
+import type { Config } from '../base/types.js';
+
+type BaseApplicationConfig = {
+  entities?: string[];
```

Comment:
Ideally this one should move to 'generator/entities' right?

Reply:
There are two differences about base-simple-application and base-application.
- supports entities in tasks
- provides full types by default.

There isn’t a good reason to move entities support to another generator.
Generator naming may be confusing, base-application is the base for a JHipster generated application (an application with entities support).